storage/redis/adapter.go

```diff
@@ -0,0 +1,101 @@
+package redis
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	r "gopkg.in/redis.v3"
+)
+
+type adapter struct {
+	bufferSize  int
+	redisClient *r.Client
+}
+
+// NewStorageAdapter returns a pointer to a new instance of a redis-based storage.Adapter.
+func NewStorageAdapter(bufferSize int) (*adapter, error) {
+	if bufferSize <= 0 {
+		return nil, fmt.Errorf("Invalid buffer size: %d", bufferSize)
+	}
+	db, err := getIntOpt("DEIS_LOGGER_REDIS_DB", 0)
```

Comment:
this func will be easier to test if this value is passed as a param. not blocking for this PR